pkg/tcpip/link/tun/device.go

```diff
@@ -264,13 +275,6 @@ func (d *Device) encodePkt(info *channel.PacketInfo) (buffer.View, bool) {
 		vv.AppendView(buffer.View(hdr))
 	}
 
-	// If the packet does not already have link layer header, and the route
-	// does not exist, we can't compute it. This is possibly a raw packet, tun
-	// device doesn't support this at the moment.
-	if info.Pkt.LinkHeader().View().IsEmpty() && len(info.Route.RemoteLinkAddress) == 0 {
```

Comment:
Why remove this?

Reply:
This code ignore packets that doesn't have Link Layer Header and does not have Link layer destination address.
This is True for every  TUN interface without packet info, therefore packets always being dropped for tun without packet info always

Reply:
Understood. We still need tests to verify that TUN with IFF_NO_PI works.

Reply:
Added the test